src/operator/nn/dnnl/dnnl_pow_mul_scalar.cc

```diff
@@ -18,51 +18,64 @@
-                                            const NDArray& input,
-                                            const NDArray& output) {
-  const NumpyBinaryScalarParam& param = nnvm::get<NumpyBinaryScalarParam>(attrs.parsed);
+bool SupportDNNLPower(const NDArray& input) {
+  return input.shape().Size() != 0 && input.shape().ndim() > 0 && input.shape().ndim() <= 6 &&
+         input.dtype() == mshadow::kFloat32;
+}
+
+DMLC_REGISTER_PARAMETER(DNNLPowMulScalarParam);
+
+DNNLPowMulScalarFwd& DNNLPowMulScalarFwd::GetCached(const DNNLPowMulScalarParam& param,
+                                                    const NDArray& input,
+                                                    const NDArray& output) {
 #if DMLC_CXX11_THREAD_LOCAL
-  static thread_local std::unordered_map<DNNLPowerSignature, DNNLPowerFwd, OpHash> fwds;
+  static thread_local std::unordered_map<DNNLPowMulScalarSignature, DNNLPowMulScalarFwd, OpHash>
+      fwds;
 #else
-  static MX_THREAD_LOCAL std::unordered_map<DNNLPowerSignature, DNNLPowerFwd, OpHash> fwds;
+  static MX_THREAD_LOCAL std::unordered_map<DNNLPowMulScalarSignature, DNNLPowMulScalarFwd, OpHash>
+      fwds;
 #endif
-  DNNLPowerSignature key;
-  key.AddSign(static_cast<float>(param.scalar));
+  DNNLPowMulScalarSignature key(param);
   key.AddSign(input);
   key.AddSign(output);
 
   auto it = fwds.find(key);
   if (it == fwds.end()) {
-    const DNNLPowerFwd fwd(input, static_cast<float>(param.scalar));
+    const DNNLPowMulScalarFwd fwd(param, input);
     it = AddToCache(&fwds, key, fwd);
   }
   return it->second;
 }
 
-DNNLPowerFwd::DNNLPowerFwd(const NDArray& input, const float exponent) {
+DNNLPowMulScalarFwd::DNNLPowMulScalarFwd(const DNNLPowMulScalarParam& param, const NDArray& input) {
   auto src_desc = input.GetDNNLData()->get_desc();
-  dnnl::eltwise_forward::desc fwd_desc(
-      dnnl::prop_kind::forward_scoring, dnnl::algorithm::eltwise_pow, src_desc, 1, exponent);
+  dnnl::eltwise_forward::desc fwd_desc(dnnl::prop_kind::forward_scoring,
+                                       dnnl::algorithm::eltwise_pow,
+                                       src_desc,
+                                       param.multiplier,
+                                       param.exponent);
   fwd_pd = std::make_shared<eltwise_fwd_pd_t>(fwd_desc, mxnet::CpuEngine::Get()->get_engine());
   fwd    = std::make_shared<eltwise_fwd_t>(*fwd_pd);
 }
 
-void DNNLPowerFwd::Execute(const NDArray& input, const OpReqType& req, const NDArray& output) {
-  auto engine           = mxnet::CpuEngine::Get()->get_engine();
-  auto src              = input.GetDNNLData();
-  dnnl_output_t out_mem = CreateDNNLMem(output, fwd_pd->dst_desc(), req, &input);
+void DNNLPowMulScalarFwd::Execute(const NDArray& input,
+                                  const OpReqType& req,
+                                  const NDArray& output) {
+  auto input_ = input;
+  if (input_.IsDNNLData())
```

Comment:
I think it shouldn't be necessary here - reordering to default format is done in DNNLRun function